lib/surveyor/models/response_set_methods.rb

```diff
@@ -185,10 +221,21 @@ def is_qualified?
       protected
 
       def dependencies(question_ids = nil)
-        question_ids = survey.sections.map(&:questions).flatten.map(&:id) if responses.blank? && question_ids.blank?
-        deps = Dependency.includes(:dependency_conditions).where({ dependency_conditions: { question_id: question_ids || responses.map(&:question_id) } })
-        # this is a work around for a bug in active_record in rails 2.3 which incorrectly eager-loads associatins when a
-        # condition clause includes an association limiter
+        if responses.blank? && question_ids.blank?
+          question_ids = survey.sections.map(&:questions).flatten.map(&:id)
+        end
+
+        deps =
+          Dependency
+            .includes(:dependency_conditions)
+            .where({
+              dependency_conditions: {
+                question_id: question_ids || responses.map(&:question_id)
+              },
+            })
+
+        # this is a work around for a bug in active_record in rails 2.3 which incorrectly
```

Comment:
Happy to know we're working around Rails 2.3...

Reply:
i wonder if ripping this out will help us...

Reply:
_Narrator: It did not help us._

Reply:
fun fact, a spec broke so leaving this for now